pkg/manager/manager.go

```diff
@@ -204,6 +212,15 @@ type Options struct {
 	// between tries of actions. Default is 2 seconds.
 	RetryPeriod *time.Duration
 
+	// OnStoppedLeadingCallback provides a way to configure a set of helper functions
+	// that are invoked when the current leader stops leading by losing the election or failing
+	// to renew the lock.
+	OnStoppedLeadingCallback []func() error
+
+	// OnNewLeaderCallback provides a way to configure a set of handlers to be invoked when the
+	// current leader changes.
+	OnNewLeaderCallback []func(string) error
```

Comment:
Why do you need the callback for OnNewLeader? Usually there is no such condition that one leader will be force migrated to another. Even after a leader lost its election, another one still has to wait for `LeaseDuration` to become a new leader.

Reply:
`OnNewLeader` is triggered when the observed state of the leader changes right ? This can handy in performing certain event driven workflow in multi node clusters. The node that becomes the leader will get a callback with the start leading callback. And when it loses, it will get an event for when the leader is lost. The `OnNewLeader` is invoked with following condition as well.
> This includes the first observed leader when the client starts.

So this can be handy to recover cases where pod crashed or restarted trying to address the Stopped leading event. It can use this call back to set the state of the world accordingly. 

Also, this provides a completeness against the Leader callback handlers.